lib/spack/spack/test/llnl/util/filesystem.py

```diff
@@ -588,3 +588,24 @@ def test_content_of_files_with_same_name(tmpdir):
     # and have not been mixed
     assert file1.read().strip() == 'file1'
     assert file2.read().strip() == 'file2'
+
+
+def test_keep_modification_time(tmpdir):
+    file1 = tmpdir.ensure('file1')
+    file2 = tmpdir.ensure('file2')
+
+    # Shift the modification time of the file 10 seconds back:
+    mtime1 = file1.mtime() - 10
+    file1.setmtime(mtime1)
+
+    with fs.keep_modification_time(file1.strpath,
+                                   file2.strpath,
+                                   'non-existing-file'):
+        file1.write('file1')
+        os.unlink(file2.strpath)
```

Comment:
Since this isn't a symlink, I think it's clearer to call `os.remove` (even though IIRC they have the same effect).